common/build-style/gem.sh

```diff
@@ -0,0 +1,125 @@
+	sed -ri "s|(\.version .*)=.*|\1 = \"${version}\"|g" ${pkgname#ruby-}.gemspec
+
+	# Replace use of `git ls-files` with `find . -type f`
+	sed -i 's|`git ls-files`|`find . -type f`|g' ${pkgname#ruby-}.gemspec
+
+	# Same as above but use -print0 when people call git ls-files -z
+	sed -i 's|`git ls-files -z`|`find . -type f`|g' ${pkgname#ruby-}.gemspec
+
+	if [ "$CROSS_BUILD" ]; then
+		# Create a .extconf file that forces the Makefile to use our environment
+		# this allows us to cross-compile like it is done with meson cross-files
+		cat>append<<EOF
+RbConfig::MAKEFILE_CONFIG['CPPFLAGS'] = ENV['CPPFLAGS'] if ENV['CPPFLAGS']
+\$CPPFLAGS = ENV['CPPFLAGS'] if ENV['CPPFLAGS']
+RbConfig::MAKEFILE_CONFIG['CC'] = ENV['CC'] if ENV['CC']
+RbConfig::MAKEFILE_CONFIG['CXX'] = ENV['CXX'] if ENV['CXX']
+RbConfig::MAKEFILE_CONFIG['LD'] = ENV['LD'] if ENV['LD']
+RbConfig::MAKEFILE_CONFIG['CFLAGS'] = ENV['CFLAGS'] if ENV['CFLAGS']
+RbConfig::MAKEFILE_CONFIG['CXXFLAGS'] = ENV['CXXFLAGS'] if ENV['CXXFLAGS']
+EOF
+
+		# Patch all instances of extconf that use create_makefile
+		for f in $(find . -type f -name 'extconf.rb'); do
+			if [ ! -f ${f}.orig ]; then
+				# Ignore extconf files that do not create makefiles
+				grep create_makefile $f || continue
+				# Create a backup which we will restore later
+				cp $f ${f}.orig
+				# Patch extconf.rb for cross compile
+				cat append ${f}.orig >> $f
+			fi
+		done
+	fi
+
+	$gem_cmd build ${pkgname#ruby-}.gemspec
+
+	if [ "$CROSS_BUILD" ]; then
+		# Restrore previous extconf.rb which we ship.
+		for f in $(find . -type f -name 'extconf.rb.orig'); do
+			mv $f ${f%.*}
+		done
+	fi
+}
+
+do_install() {
+	: ${gem_cmd:=gem}
+
+	local _GEMDIR _TARGET_PLATFORM _HOST_PLATFORM _HOST_EXT_DIR _TARGET_EXT_DIR
+
+	_GEMDIR="$($gem_cmd env gemdir)"
+
+	if [ "$CROSS_BUILD" ]; then
+		# Convert to the platforms used in our ruby packages
+		case "$XBPS_TARGET_MACHINE" in
+			x86_64) _TARGET_PLATFORM=x86_64-linux ;;
+			x86_64-musl) _TARGET_PLATFORM=x86_64-linux-musl ;;
+			i686) _TARGET_PLATFORM=i686-linux ;;
+			i686-musl) _TARGET_PLATFORM=i686-linux-musl ;;
+			armv7l) _TARGET_PLATFORM=/armv7l-linux-eabihf ;;
+			armv6l) _TARGET_PLATFORM=arm-linux-eabihf ;;
+			armv7l-musl) _TARGET_PLATFORM=armv7l-linux-musleabihf ;;
+			armv6l-musl) _TARGET_PLATFORM=arm-linux-musleabihf ;;
+			aarch64) _TARGET_PLATFORM=aarch64-linux ;;
+			aarch64-musl) _TARGET_PLATFORM=aarch64-linux-musl ;;
+		esac
+
+		case "$XBPS_MACHINE" in
+			x86_64) _HOST_PLATFORM=x86_64-linux ;;
+			x86_64-musl) _HOST_PLATFORM=x86_64-linux-musl ;;
+			i686) _HOST_PLATFORM=i686-linux ;;
+			i686-musl) _HOST_PLATFORM=i686-linux-musl ;;
+			armv7l) _HOST_PLATFORM=/armv7l-linux-eabihf ;;
+			armv6l) _HOST_PLATFORM=arm-linux-eabihf ;;
+			armv7l-musl) _HOST_PLATFORM=armv7l-linux-musleabihf ;;
+			armv6l-musl) _HOST_PLATFORM=arm-linux-musleabihf ;;
+			aarch64) _HOST_PLATFORM=aarch64-linux ;;
+			aarch64-musl) _HOST_PLATFORM=aarch64-linux-musl ;;
+		esac
+	fi
```

Comment:
Hm, shouldn't this be in `common/environment/build-style`?

Reply:
No, it is only used during build

Reply:
Ah, okay.